lib/jeos.pm

```diff
@@ -0,0 +1,60 @@
+# Copyright (C) 2015-2020 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package jeos;
+use strict;
+use warnings;
+use parent 'Exporter';
+use testapi;
+use utils 'ensure_serialdev_permissions';
+use power_action_utils 'power_action';
+use Utils::Backends 'is_hyperv';
+use version_utils qw(is_sle is_tumbleweed is_leap);
+use bootloader_setup qw(change_grub_config grep_grub_settings grub_mkconfig set_framebuffer_resolution set_extrabootparams_grub_conf);
+
+our @EXPORT = qw(expect_mount_by_uuid set_grub_gfxmode rebooting is_efi_boot);
+
+sub expect_mount_by_uuid {
+    return (is_hyperv || is_sle('>=15-sp2') || is_tumbleweed || is_leap('>=15.2'));
+}
+
+# non empty */sys/firmware/efi/* must exist in UEFI mode
+sub is_efi_boot {
+    return !script_run('test -d /sys/firmware/efi/') && script_run('test -z "$(ls -A /sys/firmware/efi/)"');
```

Comment:
this would save us from few extra `wait_serial` calls : 
```suggestion
    return !script_run('test -d /sys/firmware/efi/ && test -z "$(ls -A /sys/firmware/efi/)"');
```

Reply:
after playing a little with this in terminal I have more to say : 
```suggestion
    return script_run('test -z "$(ls -A /sys/firmware/efi/)"');
```
1. not sure why we need first check at all , second one fully cover expectations ( determine if folder exists and non-empty at the same time ). Do you bother about error message in case folder not exists ? Because it is `script_run` this will not break the test.
2. I confused with inversion `!` here . function name is `is_efi_boot` so I would expect that it returning true when `efi` folder exists and none empty , right ? But `test` returns `1` when expression is true so unless I overlook something we would get 0 when folder exists and not empty with current implementation

Reply:
we would need to invert `test -z` to `test -n`, because the directory must be non-empty.
And I am afraid we wouldn't be able to determine that the problem is missing directory or the emptiness of the directory

Reply:
1. https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/9997#discussion_r408014736 and it is in the comment above :)

2. test returns 0 when expression is true, and we need to invert it when in comes to perl

```
An exit status of zero indicates that the expression evaluated as true
       and an exit status of 1 indicates that the expression evaluated as false.
```

Reply:
And it there is not existing directory, no need to execute `ls`.

Reply:
FYI, this how fedora does it. :)
https://openqa.fedoraproject.org/tests/579214/modules/uefi_postinstall/steps/1/src

Reply:
```
➜  other test -z "$(ls -A /sys/firmware/efi/)"; echo $?
ls: cannot access '/sys/firmware/efi/': No such file or directory
0
➜  other test -z "$(ls -A ./a)"; echo $?
ls: cannot access './a': No such file or directory
0
➜  other mkdir a
➜  other test -z "$(ls -A ./a)"; echo $?
0
➜  other mkdir a/b                      
➜  other test -z "$(ls -A ./a)"; echo $?
1
➜  other test -n "$(ls -A ./a)"; echo $?
0
```

Reply:
yup, the *error* message is not nice there

Reply:
>     2. test returns 0 when expression is true, and we need to invert it when in comes to perl
> 
> 
> ```
> An exit status of zero indicates that the expression evaluated as true
>        and an exit status of 1 indicates that the expression evaluated as false.
> ```

from where you quoting this ? Just run this on my laptop : 
```
13:38-e113$[0] test -z "$(ls -A /home/asmorodskyi/)"
15:32-e113$[0] echo $?
1
```

Reply:
after posting this I realized that you , right . I just confuse what is expected here , sorry